src/app/pages/ConsensusBlocksPage/index.tsx

```diff
@@ -47,25 +50,29 @@ export const ConsensusBlocksPage: FC = () => {
     {
       limit: tableView === TableLayout.Vertical ? offset + PAGE_SIZE : PAGE_SIZE,
       offset: tableView === TableLayout.Vertical ? 0 : offset,
+      before: enablePolling ? undefined : beforeDate,
```

Comment:
Looks like before is an exclusive `<` operation, so we are off by a few:
- load 1st page https://nexus.oasis.io/v1/sapphire/transactions?limit=10&offset=0
```
[
    "d358dcd4d3d347645b243a36c83bded7d1eb6ee2551111c22c3fcb5ee4c3a0aa",
    "7e18f5d3aa89d84ba987316e721dbf3a5b917017ae73260533cd561d08c3e9a1",
    "01bb606eba3c2d5bb6e1e1176328fa44a8b2e66d7349f3a43bbb83e32be83456",
    "c790cea64daa8d025d168de9de3522f56c4d2907c1dc75fc57f976f82cc3feda",
    "cb1e1673b7d941803a9cb131305145072e9b05f8825a256b4d36999756048afb",
    "cc551595d2daca72c7e841cab48b8a8aae2ca207018ce35deeb79873143aebfa",
    "fc031817e89c59b8bfad6e01411e914976520ddf2c9eddef6833f41e48bda4a4",
    "87c40cf65bc7d3d1e8bc6dd30907ce758312789fb8a6dae7182af589c158824e",
    "9e439631945a5f5bd4ca07dd994304d3d1eb2a50ba17069abaaca4e80719fd00",
    "4321edaf10a99f37a294c64784f8c2d8e36d72bb9340b4918e35517f3ec4a38c"
]
```
- first result is `"timestamp": "2024-09-18T23:16:52Z"`
- load 2nd page https://nexus.oasis.io/v1/sapphire/transactions?limit=10&offset=10&before=2024-09-18T23:16:52Z
```
[
    "7bb006068684f7a60b534fbeabf3597e3b112cbf9aee61ad5be06d3a9c3afdcc",
    "8841255abaefeb78d614b4af24a6eeeff10f7bca92fd3f17958372268a324fec",
    "cd60a60b04d42479fd955befdf1f1b263646b1f5b96b7bb01fc26b457c400bde",
    "81d429e9caa4e039db9c75a0c9ea7342cd6cc03a982ac67c354473ad614d123b",
    "2faba905c1aa0e2ecf84de2786c5cd02c1fd1af8bc99102e5be38f5e36f58383",
    "d1f02191fa5f013b3a214127fc0c1aa9e21080fd9f66ce583af990e3f484d18a",
    "81ad1268311659ed22f0964fb00bd54929717b07ff0d88a9075a97664703d839",
    "4025c7cfb0c76bac6f1d30a14552f2306b73cbc80c6f19142b8dd6145243f4fe",
    "16927fb4d2b5658ab7125880d83562e7f684e85f6cc3e01bfc16261263a05688",
    "82f5a884804ba5d41c9edd352a623aa17872fdf479efa79eed3c9810b2a0bc47"
]
```
- this actually skipped one transaction (luckily only one tx had `23:16:52` timestamp)

A simple way to confirm it
https://nexus.oasis.io/v1/sapphire/transactions?limit=10&offset=0&before=2024-09-18T23:16:52Z
first result is `2024-09-18T23:16:34Z` :scream:
```
[
    "7e18f5d3aa89d84ba987316e721dbf3a5b917017ae73260533cd561d08c3e9a1",
    "01bb606eba3c2d5bb6e1e1176328fa44a8b2e66d7349f3a43bbb83e32be83456",
    "c790cea64daa8d025d168de9de3522f56c4d2907c1dc75fc57f976f82cc3feda",
    "cb1e1673b7d941803a9cb131305145072e9b05f8825a256b4d36999756048afb",
    "cc551595d2daca72c7e841cab48b8a8aae2ca207018ce35deeb79873143aebfa",
    "fc031817e89c59b8bfad6e01411e914976520ddf2c9eddef6833f41e48bda4a4",
    "87c40cf65bc7d3d1e8bc6dd30907ce758312789fb8a6dae7182af589c158824e",
    "9e439631945a5f5bd4ca07dd994304d3d1eb2a50ba17069abaaca4e80719fd00",
    "4321edaf10a99f37a294c64784f8c2d8e36d72bb9340b4918e35517f3ec4a38c",
    "d2d42ed7d48abd5b4a7c29dd2414d8c61f20a92fdaa8bd01df3e76126c6aab02"
]
```

alternatively +1 second does return the original set
https://nexus.oasis.io/v1/sapphire/transactions?limit=10&offset=0&before=2024-09-18T23:16:53Z
```
[
    "d358dcd4d3d347645b243a36c83bded7d1eb6ee2551111c22c3fcb5ee4c3a0aa",
    "7e18f5d3aa89d84ba987316e721dbf3a5b917017ae73260533cd561d08c3e9a1",
    "01bb606eba3c2d5bb6e1e1176328fa44a8b2e66d7349f3a43bbb83e32be83456",
    "c790cea64daa8d025d168de9de3522f56c4d2907c1dc75fc57f976f82cc3feda",
    "cb1e1673b7d941803a9cb131305145072e9b05f8825a256b4d36999756048afb",
    "cc551595d2daca72c7e841cab48b8a8aae2ca207018ce35deeb79873143aebfa",
    "fc031817e89c59b8bfad6e01411e914976520ddf2c9eddef6833f41e48bda4a4",
    "87c40cf65bc7d3d1e8bc6dd30907ce758312789fb8a6dae7182af589c158824e",
    "9e439631945a5f5bd4ca07dd994304d3d1eb2a50ba17069abaaca4e80719fd00",
    "4321edaf10a99f37a294c64784f8c2d8e36d72bb9340b4918e35517f3ec4a38c"
]
```

Reply:
Not sure if +1 second always returns the original set. If 10 transactions have 23:16:52 timestamp, will they all have been processed together when loading the first page, or can we get 2/10

Reply:
I did not catch this in docs `A filter on maximum transaction time, exclusive`.  Was chatting with backend and 

- Adding 1s on frontend _"probably will be fine since all transactions have the same timestamp of the block and there is less than one block per second"_

- request API change to adjust before filters (seems easy) https://github.com/oasisprotocol/nexus/blob/main/storage/client/queries/queries.go#L529) but involves Nexus team agreement